rpe/resources/gcp.py

```diff
@@ -1114,3 +1083,73 @@ def _get_request_args(self):
                 self._resource_data["name"],
             ),
         }
+
+
+class GcpDataformRepository(GoogleAPIResource):
+    service_name = "dataform"
+    resource_path = "projects.locations.repositories"
+    version = "v1beta1"
+
+    required_resource_data = ["name", "location", "project_id"]
+
+    resource_components = {
+        "iam": "getIamPolicy",
+    }
+
+    resource_type = "dataform.googleapis.com/Repository"
```

Comment:
this seems to be our naming convention, can we add some documentation to describe the purpose of this field and where it needs to tie up for our engine? this also gets passed in differently from cai event - asset_type?

Reply:
Cai events use the from_cai_data method, and that method uses a helper method, which ties up the asset_type with the resource_type of the Class to get the right Resource

Yes, this probably requires better documentation, I think it's better to make a separate internal confluence detailing how rpe-lib and real time enforcer is used with the safeguards engine

Reply:
we can start with some comments in code, confluence is always a great addition with more details..